clients/hadoopfs/src/main/java/io/lakefs/MetadataClient.java

```diff
@@ -23,7 +22,7 @@ public class MetadataClient {
 
     public MetadataClient(FileSystem fs) {
         if (fs == null) {
-            throw new NullArgumentException("fs");
+            throw new java.lang.IllegalArgumentException();
```

Comment:
This loses information (that `fs == null`).  Can you either revert or add that information in the message?